integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/generator.py

```diff
@@ -72,11 +73,21 @@ class AmazonBedrockGenerator:
     def __init__(
         self,
         model: str,
-        aws_access_key_id: Optional[str] = None,
-        aws_secret_access_key: Optional[str] = None,
-        aws_session_token: Optional[str] = None,
-        aws_region_name: Optional[str] = None,
-        aws_profile_name: Optional[str] = None,
+        aws_access_key_id: Optional[Secret] = EnvVarSecret(env_vars=["AWS_ACCESS_KEY_ID"], strict=False).from_env_var(
+            "AWS_ACCESS_KEY_ID"
+        ),
+        aws_secret_access_key: Optional[Secret] = EnvVarSecret(
+            env_vars=["AWS_SECRET_ACCESS_KEY"], strict=False
+        ).from_env_var("AWS_SECRET_ACCESS_KEY"),
+        aws_session_token: Optional[Secret] = EnvVarSecret(env_vars=["AWS_SESSION_TOKEN"], strict=False).from_env_var(
+            "AWS_SESSION_TOKEN"
+        ),
+        aws_region_name: Optional[Secret] = EnvVarSecret(env_vars=["AWS_DEFAULT_REGION"], strict=False).from_env_var(
+            "AWS_DEFAULT_REGION"
+        ),
```

Comment:
Neither `EnvVarSecret` nor `TokenSecret` are meant to be used directly. Let's go with this:
```python
Secret.from_evn_var(["env1", "env2"], strict=False)
```

Reply:
thanks 👍🏽